src/backend/distributed/README.md

```diff
@@ -0,0 +1,2079 @@
+
+All Citus table types have the notion of a “shard”, though in many cases there is only a single shard.  
+
+**Shard**: The table that contains the actual data in a Citus table. Shards reside in the same schema as the Citus table and are named after the Citus table with the shard ID as a suffix. Shards are hidden from catalog queries unless you SET citus.override_table_visibility TO off. Hash-distributed tables have multiple shards which each have a distinct shardminvalues/shardmaxvalue. 
+
+**Colocation group**: A set of distributed tables that have the same co-location ID, such that their shards are always co-located. 
+
+**Shard group**: A set of shards with the same shardminvalue/shardmaxvalue that are part of the same co-location group. Citus guarantees that shards from the same shard group are always placed on the same node group. 
+
+**Shard placement**: The assignment of a shard to a node group. There can be multiple placements of the same shard if the table is replicated (e.g. reference tables). 
+
+**Shard placement group**: The assignment of a shard to a node group must be the same for all shards in a shard group, since those are always co-located. We’ll refer to the group of placements of a shard group as a shard placement group. A single shard group can have multiple shard placement groups. 
+
+**Node**: A single PostgreSQL/Citus server listed in pg_dist_node and added via SELECT citus_add_node(..). 
+
+**Node group**: Each primary node can have 0 or more physical replicas in read replica clusters. Together the nodes form a node group identified by the groupid in pg_dist_node. Per convention, the coordinator(s) have group ID 0. Each node can know its own node groupid by reading it from pg_dist_local_group. 
+
+**Coordinator**: The node with groupid 0, which can perform reads, writes, and administrative operations such as adding a node, rebalancing, and schema changes. 
+
+**Worker nodes**: Nodes with groupid > 0, which can perform reads and writes, but not administrative operations.  
+
+**Cluster**: The combination of worker nodes and coordinator is a cluster. When the cluster has a read replica, the nodes in the read replica are listed in pg_dist_node with a different nodecluster value, and the servers have a corresponding citus.cluster_name in their postgresql.conf. That way, nodes know which other nodes in pg_dist_node belong to their cluster, and they will ignore others. 
+
+**Client connections**: Connections made by the client to any of the nodes in the cluster. Each client connection is backed by a postgres process/backend, which we sometimes refer to as a client session. 
+
+**Internal connections**: Connections to other nodes made by a client session. Each internal connection is backed by a process, which we sometimes refer to as an internal session. In the code, you can use IsCitusInternalBackend() 
```

Comment:
Possibly not relevant here. In the case where a client connection is made directly to the node where data resides, what determines whether an internal connection is created (i.e. PG connecting to itself) vs. doing all the computation in the client connection?

Reply:
the local execution section has more details on this